tools/scripts/release-cleanup.ps1

```diff
@@ -0,0 +1,108 @@
+﻿# Copyright (c) Microsoft. All rights reserved.
+# Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#This script assumes that the OctoKit Nuget package is a custom install and is in the current folder.
+
+$deleteList = @()
+
+# Extracts a version from the release name.
+function Extract-Version($specificRelease){
+    $versionString = $specificRelease.Name
+    if($versionString -match ".*v(\d+\.\d+\.\d+\.\d+).*"){
+        return $matches[1]
+    } else {
+        $deleteList += $specificRelease
+        return [string]::Empty
+    }
+}
+
+# Sorts the releases based on the version number
+function Sort-Releases($releases){
+    $releaseMap = @{}
+    foreach($release in $releases.Result){
+        $releaseVersion = [System.version](Extract-Version $release)
+        if(-not [string]::IsNullOrEmpty($releaseVersion)){
+            $releaseMap.Add($releaseVersion,$release)
+        }
+    }
+
+    $releaseMap = $releaseMap.GetEnumerator() | Sort-Object -Descending -Property Name
+    return $releaseMap
+}
+
+# Delete a release
+function Delete-Release($release){
+    # Null checks
+    if([string]::IsNullOrEmpty($release.Id)){
+        Write-Host "No Release Id found"
+        return;
+    }
+
+    Write-Host "Deleting release" + $release.Name
+    $response = $instance.Repository.Release.Delete($repoId, $release.Id);
```

Comment:
>Release [](start = 37, length = 7)

Does this also delete the tag, or is that something that we need to add?

Reply:
When we delete a release (even from the UI) we do not delete the tag that the release was created at.

What that means is that the MSI will be deleted however you can still download the code from which the MSI was created. 

Git does not have an API to delete tags -  https://developer.github.com/v3/git/tags/

What that means to us is that after a release is deleted we still see the tag in the release view. Personally I feel that this should be okay (and I would actually advocate that we keep these tags) as they help us identify the cut points in the code for these releases

Reply:
Discussed offline.